python_benchmarks/core.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import ctypes
+import torch
+from torch.profiler import profile, ProfilerActivity
+
+libnames = ("libcuda.so", "libcuda.dylib", "nvcuda.dll", "cuda.dll")
+for libname in libnames:
+    try:
+        cuda = ctypes.CDLL(libname)
+    except OSError:
+        continue
+    else:
+        break
+else:
+    raise OSError("could not load any of: " + " ".join(libnames))
+
+CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE = 38
+l2_cache_size = ctypes.c_int()
+device = torch.cuda.current_device()
+cuda.cuDeviceGetAttribute(
+    ctypes.byref(l2_cache_size), CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE, device
+)
+
+L2_CACHE_SIZE = l2_cache_size.value
+
+
+def clearL2Cache():
+    n_elements = L2_CACHE_SIZE // 4
+    x = torch.empty(n_elements, dtype=torch.float32, device="cuda", requires_grad=False)
+    y = torch.clone(x)
+
+
+class NVFBenchmark(object):
+    def __init__(self, benchmark_fixture, precision: float = 1e-6):
+        # Initialize a Torch Profiler object
+        self.prof = profile(activities=[ProfilerActivity.CUDA, ProfilerActivity.CPU])
```

Comment:
It'd be great to add some type hints: https://docs.python.org/3/library/typing.html. For example, it's hard for me to figure out the type of `self.prof` which type hints help immediately.

Reply:
Added details for class members in the docstring for the class.